src/test/java/mocks/MockRepository.java

```diff
@@ -0,0 +1,90 @@
+package mocks;
+
+import http_standards.MIMETypes;
+import repository.MockFile;
+import repository.Repository;
+
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class MockRepository implements Repository  {
+    String directory;
+    private HashMap<String, MockFile> memoryRepository = new HashMap<>();
+
+    public MockRepository(String directory) {
```

Comment:
I know you implemented this in this way so that you wouldn't have to rewrite a bunch of tests (since they depended on the public directory containing these files), but you might find that adding other constructors to allow the user to create a mock repository with certain directories/files will make your existing tests clearer. With this implementation, the contents of the mock repository are hidden in this constructor.

Reply:
I've changed the mock repository to instantiate with a empty repository. From there, users can write new files using `writeFile(intended file path, file type, file contents)`